olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Messages.h

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <boost/optional.hpp>
+#include <olp/dataservice/read/DataServiceReadApi.h>
+#include <olp/dataservice/read/model/Data.h>
+#include <olp/dataservice/read/model/StreamOffsets.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+namespace model {
+
+class DATASERVICE_READ_API Metadata final {
+ public:
+  const std::string& GetPartition() const { return partition_; }
+  void SetPartition(std::string value) { partition_ = std::move(value); }
+
+  const boost::optional<std::string>& GetChecksum() const { return checksum_; }
+  void SetChecksum(std::string value) { checksum_ = std::move(value); }
+
+  const boost::optional<int64_t>& GetCompressedDataSize() const {
+    return compressed_data_size_;
+  }
+  void SetCompressedDataSize(int64_t value) { compressed_data_size_ = value; }
+
+  const boost::optional<int64_t>& GetDataSize() const { return data_size_; }
+  void SetDataSize(int64_t value) { data_size_ = value; }
+
+  const Data& GetData() const { return data_; }
+  void SetData(Data value) { data_ = std::move(value); }
+
+  const boost::optional<std::string>& GetDataHandle() const {
+    return data_handle_;
+  }
+  void SetDataHandle(std::string value) { data_handle_ = std::move(value); }
+
+  const boost::optional<int64_t>& GetTimestamp() const { return timestamp_; }
+  void SetTimestamp(int64_t value) { timestamp_ = value; }
+
+ private:
+  /// A key that specifies which
```

Comment:
As this is a public class, I would rather put these comments on top of the above getters.